### PR TITLE
Symmetric Grid Updates

### LINKED ISSUE #41: Detect symmetric mode for MOC script
An extra land point over Antarctica appears when running the model in symmetric memory mode.  A check for this condition needs to be added.

### PR DESCRIPTION
**Pull request by jkrasting**
- Updates to MOC and heat transport diagnostics
- Combines #60 into one PR

Closes #41

### CODE REVIEW

File: om4labs/om4common.py [hunk 1]
```diff
@@ -261,13 +264,21 @@ def grid_from_supergrid(ds, point_type="t"):
...
     """
-    geolat = subsample_supergrid(ds, "y", point_type)
-    geolon = subsample_supergrid(ds, "x", point_type)
-    area = sum_on_supergrid(ds, "area", point_type)
+
+    geolat = subsample_supergrid(ds, "y", point_type, outputgrid=outputgrid)
+    geolon = subsample_supergrid(ds, "x", point_type, outputgrid=outputgrid)
+    area = sum_on_supergrid(ds, "area", point_type, outputgrid=outputgrid)
+
+    # kludge - requires fix in static_downsampler pkg
```
Comment (raphaeldussin) [inline om4labs/om4common.py:272]: should not be necessary anymore

File: om4labs/om4common.py [hunk 2]
```diff
@@ -462,10 +483,26 @@ def read_topography(dictArgs):
...
+
+        # southern boundary
+        south = xr.DataArray(tmp[0, :], dims=("x"))
+        inner_w_poles = xr.concat([south, inner], dim="y")
+
+        # apply E-W boundary condition
+        west = xr.DataArray(inner_w_poles[-1, :], dims=("y"))
+        out = xr.concat([west, inner_w_poles], dim="x")
+        depth = out
+        depth = depth[:, 1:]
```
Comment (raphaeldussin) [inline om4labs/om4common.py:504]: I don't understand why using periodicity to then exclude eastern-most point

**Review by raphaeldussin — CHANGES_REQUESTED** [review]

File: om4labs/om4common.py (hunk 2 above)
Comment (jkrasting) [reply om4labs/om4common.py:504]: We need a depth field with the same dimensions as the area field.  In the case of a symmetric grid, we need to add a row and column.

Comment (raphaeldussin) [reply om4labs/om4common.py:504]: yes but you're removing said column on line 504. Also if you want to plot the depth on MOC plots (V-point)
we should use xgcm to interpolate depth from T to V point

Comment (jkrasting) [reply om4labs/om4common.py:500]: Ah. Good point. (no pun intended)

Comment (raphaeldussin) [reply om4labs/om4common.py:500]: how about you build the extended depth array and then each diag will interp or extract according to the grid point type used

File: om4labs/om4common.py [hunk 3]
```diff
@@ -462,10 +479,26 @@ def read_topography(dictArgs):
...
+        depth = ds.depth
+
+    if outputgrid == "symetric":
+        """ extend array, assuming periodic/tripolar grid"""
+        warnings.warn("assuming the grid is periodic and tripolar")
+        tmp = depth.values
+        inner = xr.DataArray(tmp, dims=("y", "x"))
+
+        # southern boundary
+        south = xr.DataArray(tmp[0, :], dims=("x"))
```
Comment (raphaeldussin) [inline om4labs/om4common.py:493]: you can also use this routine: https://github.com/raphaeldussin/static_downsampler/blob/3f7c0354e907cb420ff67bc97c6f4c053b3b0715/static_downsampler/static.py#L185

**Review by raphaeldussin — APPROVED** [review]

**Pull request merged**